app/models/subscription.rb

```diff
@@ -0,0 +1,39 @@
+class Subscription < ApplicationRecord
+  
+  belongs_to :owner, class_name: :User
+  belongs_to :plan
+  has_and_belongs_to_many :users, before_add: :check_limits
+
+  before_create :check_active_subscriptions
+  before_create :set_expiry_date
+
+  after_create_commit :add_owner_to_users
+
+  private def check_active_subscriptions
+    if owner.active_subscription?
```

Comment:
consider the scenario
user has a week left in the subscription. You send a reminder email. He goes and purchases the membership. Will he be able to do so? If not, why? Should he be able to do so? If yes, how?

Reply:
Sir, now it is fixed.
We have start date and end date.
User has subscribe for start date > end_date of current subscriptions.